src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    private final SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] checkLine;
+        StringBuilder finalReport = new StringBuilder();
+        finalReport.append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo).append("\r");
+
+        for (String name: names) {
+            finalReport.append("\n").append(name).append(" - ");
+            int salary = 0;
+
+            for (String dateString : data) {
+                checkLine = splitString(dateString, " ");
+                if (name.equals(checkLine[1])) {
+                    if (isDateInRange(dateFrom, dateTo, checkLine[0])) {
+                        salary = salary
+                                + (Integer.parseInt(checkLine[2]) * Integer.parseInt(checkLine[3]));
+                    }
+                }
+            }
+
+            if (!name.equals(names[names.length - 1])) {
+                finalReport.append(salary).append("\r");
+            } else {
+                finalReport.append(salary);
+            }
+
+        }
+
+        return finalReport.toString();
+    }
+
+    public boolean isDateInRange(String dateRangeBegin, String dateRangeEnd, String dataSalary) {
```

Comment:
I don't think you need whole method for dealing with dates. When you will switch to LocalDate it should be simple enough and no exception handling is needed